group/basic/main.tf

```diff
@@ -1,7 +1,8 @@
 # AWS Autoscaling group
 
 ## Creates security group
-resource "aws_security_group" "sg_asg" {
+resource "aws_security_group" "asg-sg" {
+  name_prefix = "${var.stack_item_label}-"
```

Comment:
I wanted to see what this actually results in. I've not tried it just yet, but generally agree with your recommendation.